pkg/cmd/controller/unidling.go

```diff
@@ -42,5 +43,6 @@ func RunUnidlingController(ctx *ControllerContext) (bool, error) {
 
 	go controller.Run(ctx.Stop)
 
+	unidlingMetrics.InitializeUnidlingControllerMetrics()
```

Comment:
I think you'll want to initialize the metric before you run the controller in a goroutine. Othwerwise you risk a race condition incrementing the metric.